src/csharp/Microsoft.Spark/Sql/Types/DataType.cs

```diff
@@ -160,6 +160,14 @@ internal static DataType ParseDataType(JToken json)
                     }
                     else if (typeName == "udt")
                     {
+                        if (typeJObject.TryGetValue("class", out JToken classToken))
+                        {
+                            if (typeJObject.TryGetValue("sqlType", out JToken sqlTypeToken))
+                            {
+                                return new StructType(sqlTypeToken as JObject);
```

Comment:
`return new StructType((JObject)sqlTypeToken);` so that we get a better error message.